controllers/operator_controller.go

```diff
@@ -277,3 +287,51 @@ func (r *OperatorReconciler) existingBundleDeploymentUnstructured(ctx context.Co
 	}
 	return &unstructured.Unstructured{Object: unstrExistingBundleDeploymentObj}, nil
 }
+
+// verifyBDStatus reads the various possibilities of status in bundle deployment and translates
+// into corresponding operator condition status and message.
+func verifyBDStatus(dep *rukpakv1alpha1.BundleDeployment) (metav1.ConditionStatus, string) {
+	isValidBundleCond := apimeta.FindStatusCondition(dep.Status.Conditions, rukpakv1alpha1.TypeHasValidBundle)
+	isInstalledCond := apimeta.FindStatusCondition(dep.Status.Conditions, rukpakv1alpha1.TypeInstalled)
+
+	if isValidBundleCond == nil && isInstalledCond == nil {
+		return metav1.ConditionUnknown, fmt.Sprintf("waiting for bundleDeployment %q status to be updated", dep.Name)
+	}
+
+	if isValidBundleCond != nil && isValidBundleCond.Status == metav1.ConditionFalse {
+		return metav1.ConditionFalse, isValidBundleCond.Message
+	}
+
+	if isInstalledCond != nil && isInstalledCond.Status == metav1.ConditionFalse {
+		return metav1.ConditionFalse, isInstalledCond.Message
+	}
+
+	if isInstalledCond != nil && isInstalledCond.Status == metav1.ConditionTrue {
+		return metav1.ConditionTrue, "install was successful"
+	}
+	return metav1.ConditionUnknown, fmt.Sprintf("could not determine the state of bundleDeployment %s", dep.Name)
+}
+
+// mapBDStatusToReadyCondition returns the operator object's "TypeReady" condition based on the bundle deployment statuses.
+func mapBDStatusToReadyCondition(existingBD *rukpakv1alpha1.BundleDeployment, observedGeneration int64) metav1.Condition {
+	// update operator status:
+	// 1. If the Operator "Ready" status is "Unknown": The status of successful bundleDeployment is unknown, wait till Rukpak updates the BD status.
+	// 2. If the Operator "Ready" status is "True": Update the "successful resolution" status and return the result.
+	// 3. If the Operator "Ready" status is "False": There is error observed from Rukpak. Update the status accordingly.
+	status, message := verifyBDStatus(existingBD)
+	var reason string
+	// TODO: introduce a new reason for condition Unknown, instead of defaulting it to Installation Succeeded.
+	if status == metav1.ConditionTrue {
+		reason = operatorsv1alpha1.ReasonInstallationSucceeded
+	} else {
+		reason = operatorsv1alpha1.ReasonBundleDeploymentFailed
```

Comment:
For follow-up: don't leak rukpak names in Operator API (i.e. change this to "InstallationFailed")